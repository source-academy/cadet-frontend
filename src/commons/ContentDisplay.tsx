import { Card, Elevation } from '@blueprintjs/core';
import * as React from 'react';

export type ContentDisplayProps = {
  fullWidth?: boolean;
  display: JSX.Element;
  loadContentDispatch: () => void;
};

class ContentDisplay extends React.Component<ContentDisplayProps, {}> {
  public componentDidMount() {
    this.props.loadContentDispatch();
  }

  public render() {
    return (
      <div className="ContentDisplay row center-xs">
        <div
          className={`${
            this.props.fullWidth ? 'col-md-12' : 'col-md-10 col-xs-11'
          } contentdisplay-content-parent`}
        >
          <Card className="contentdisplay-content" elevation={Elevation.THREE}>
            {this.props.display}
          </Card>
        </div>
      </div>
    );
  }
}

export default ContentDisplay;
