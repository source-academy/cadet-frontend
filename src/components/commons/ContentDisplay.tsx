import { Card } from '@blueprintjs/core'
import * as React from 'react'

export interface IContentDisplayProps {
  display: JSX.Element
  loadContentDispatch: () => void
}

class ContentDisplay extends React.Component<IContentDisplayProps, {}> {
  public componentDidMount() {
    this.props.loadContentDispatch()
  }

  public render() {
    return (
      <div className="ContentDisplay row center-xs">
        <div className="col-xs-10 contentdisplay-content-parent">
          <Card className="contentdisplay-content" elevation={1}>
            {this.props.display}
          </Card>
        </div>
      </div>
    )
  }
}

export default ContentDisplay
