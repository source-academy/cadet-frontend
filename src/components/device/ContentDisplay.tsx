import * as React from 'react'

export interface IContentDisplayProps {
  displayStream: () => JSX.Element
  loadContentDispatch: () => void
}

class ContentDisplay extends React.Component<IContentDisplayProps, {}> {
  public componentDidMount() {
    this.props.loadContentDispatch()
  }

  public render() {
    return (
      <div className="Announcements row center-xs">
        <div className="col-xs-10">{this.props.displayStream()}</div>
      </div>
    )
  }
}

export default ContentDisplay
