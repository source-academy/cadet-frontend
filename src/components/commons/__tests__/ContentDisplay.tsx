import { mount } from 'enzyme';
import * as React from 'react';

import ContentDisplay, { IContentDisplayProps } from '../ContentDisplay';

const mockProps: IContentDisplayProps = {
  display: <div> Test Content </div>,
  loadContentDispatch: () => {}
};

test('ContentDisplay page renders correctly', () => {
  const app = <ContentDisplay {...mockProps} />;
  const tree = mount(app);
  expect(tree.debug()).toMatchSnapshot();
});
