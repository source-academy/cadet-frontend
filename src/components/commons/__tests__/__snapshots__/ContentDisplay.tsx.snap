// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentDisplay page renders correctly 1`] = `
"<ContentDisplay display={{...}} loadContentDispatch={[Function: loadContentDispatch]}>
  <div className=\\"ContentDisplay row center-xs\\">
    <div className=\\"col-xs-10 contentdisplay-content-parent\\">
      <Blueprint3.Card className=\\"contentdisplay-content\\" elevation={3} interactive={false}>
        <div className=\\"bp3-card bp3-elevation-3 contentdisplay-content\\">
          <div>
             Test Content 
          </div>
        </div>
      </Blueprint3.Card>
    </div>
  </div>
</ContentDisplay>"
`;
